### ISSUE TITLE
Smoke tests

### ISSUE THREAD

**Issue by jayvdb** [bartTC/pip-check#26]
It would be useful to have some basic sanity/smoke tests to verify that the package works.  All pip related tools tend to be brittle, due to the way pip is developed/maintained.
e.g. I have created https://build.opensuse.org/package/show/home:jayvdb:py-new/python-pip-check , but I need some way to verify it works with the openSUSE Tumbleweed version of pip, otherwise I cant submit it to the openSUSE official collection of packages.

**Comment by bartTC** [comment]
@jayvdb what do you miss with the tests which are in place now? They are very basic indeed but should cover what you describe here.

To trigger the tests just run tox:

```
$ tox
```

**Comment by jayvdb** [comment]
In an rpmbuild environment, tox is hard to use.
I'll try copying those commands over into my `%check` section

**Comment by jayvdb** [comment]
Ok, done that.  Only `--help` works, because the VM is isolated from the internet.  The tests need to use pre-prepared test data.  Often pip tools use a fake pypi server to perform the testing.

**Comment by jayvdb** [comment]
Worth also noting that those commands are failing but the exit status is 0.

**Comment by bartTC** [comment]
I dont see that, whats your output?

**Comment by jayvdb** [comment]
You can see the output by clicking on the 'successful' at https://build.opensuse.org/package/show/home:jayvdb:py-new/python-pip-check ; might need to log in though.

The output is the usual pip failure-retrying output as it tries to reach a server when there is no network; easy to reproduce by unplugging from the internet.

**Comment by bartTC** [comment]
Interesting! I never ran into this. I made some improvements in that direction:

- If the `pip` command fails generally with a status code 1+ it will show so.
- If there are network issues it will fetch it and show an error as well.
- I limit the number of retries to 1 so it will fail very fast with network issues.

I had trouble with this because pip is happily sending a 0 exit code despite it spits your screen full of errors and warnings. I made them an issue for it: https://github.com/pypa/pip/issues/7470

```
❯ pip-check -H
Loading package versions...

pip indicated that it has connection problems. Please check your network.%

❯ echo $?
1
```
